Source/WebCore/dom/Document.cpp

```diff
@@ -8906,12 +8906,16 @@ void Document::handlePopoverLightDismiss(const PointerEvent& event, Node& target
             auto isShowingAutoPopover = [](HTMLElement& element) -> bool {
                 return element.popoverState() == PopoverState::Auto && element.popoverData()->visibilityState() == PopoverVisibilityState::Showing;
             };
-            for (auto& element : lineageOfType<HTMLElement>(*startElement)) {
-                if (!clickedPopover && isShowingAutoPopover(element))
-                    clickedPopover = &element;
+            for (auto* element = startElement; element; element = element->parentElementInComposedTree()) {
+                if (!is<HTMLElement>(element))
+                    continue;
+
+                auto* htmlElement = downcast<HTMLElement>(element);
```

Comment:
Two choices that are better style: use `dynamicDowncast` and check for `nullptr` rather than using both is and `downcast`, or use `is` and `downcast`, but pass a reference instead of a pointer to `downcast`.